### PR TITLE
remove frame.__finalize__, ajust test

### PR DESCRIPTION
**Pull request by thpetsen**

### CODE REVIEW

**Comment by JanusWesenberg** [comment]
Could you maybe provide a bit of context for why you want to rip out the entire metadata propagation?

**Comment by thpetsen** [comment]
1) The current implementation does not seem to provide any additional functionality to `base.__finalize__`
  a) If the current implementation does provide functionality, this should be demonstrated and verified by specific tests (Factory Acceptance Test), tests that demonstrates that the software performs as defined and expected.

2) The default implementation seems pretty generic, seems to cover all inputs (no warnings in jacktop):
```
    if isinstance(other, NDFrame):
        for name in other.attrs:
            self.attrs[name] = other.attrs[name]
        # For subclasses using _metadata.
        for name in self._metadata:
            assert isinstance(name, str)
            object.__setattr__(self, name, getattr(other, name, None))
    return self
```

3) When using pdtable in jacktop, we discovered that a lot of "method" was missing from the implementation
When reading up on the documentation for these methods, I discovered a warning stating:
```
  .. warning:

     The value passed as `method` are not currently considered
     stable across pandas releases.
```

Indicating that we should probably not override this functionality, and it is specifically this `method` argument that raised the warnings in jacktop.

4) Removing the` __finalize__ `resulted in only one minor error in our tests (failure to rase a specific excepton on illegal input)
  This fuctionality does not seem critical to the intended use of metadata.

5) I did test propagating` __finalize__` to base class` __finalize__ `on the missing "method" ending up removing it entirely since it did not seem to do anything.
  At least not detectable by any of our tests.

6) If we kept the previous `__finalize__` we needed to implement all the `"method"` cases ourselves.
First day of jacktop the methods:` take, sort_values, replace, reindex, merge `showed up.
I.e. we can not guarantee that pdtable could be used unless we implement all the methods listed in the [frame documentation](https://pandas.pydata.org/docs/reference/frame.html).

**Comment by jfcorbett** [comment]
@JanusWesenberg You originally wrote the `__finalize__()` override method, I'm sure you had a reason...
Perhaps you could give us an example where we see it in action?

File: pdtable/table_metadata.py [hunk 1]
```diff
@@ -51,6 +51,9 @@ class TableMetadata:
     origin: Optional[
         str
     ] = ""  # Should be replaced with a TableOrigin object to allow file-edit access
```
Comment (JanusWesenberg) [inline pdtable/table_metadata.py:53]: What is the intended use for `col_specs`?

**Comment by JanusWesenberg** [comment]
>     1. The current implementation does not seem to provide any additional functionality to `base.__finalize__`
>        a) If the current implementation does provide functionality, this should be demonstrated and verified by specific tests (Factory Acceptance Test), tests that demonstrates that the software performs as defined and expected.

I agree this is not covered by tests, since it is still copied directly from my original POC-code, and was just intended to test that we could propagate metadata across dataframe operations.

>     2. The default implementation seems pretty generic, seems to cover all inputs (no warnings in a project we're putting pdtable in):

The reason I added this override is that the stock implementation does not combine metadata from all sources on join and concat operations. 

>     3. If we kept the previous `__finalize__` we needed to implement all the `"method"` cases ourselves.
>        First day of using pdtable in the project mentioned above, the methods: `take, sort_values, replace, reindex, merge`showed up. 

Overriding `__finalize__`  is not discussed in the docs (although I am pretty sure it was when I made the POC, with the caveat that method was undefined). I was lead to this approach by a SO post by a core pandas contributor pointing to the implementation in [geopandas](https://github.com/geopandas/geopandas/blob/924cdf65c7c15b01749d1cdd036c5c291e87b0f4/geopandas/geodataframe.py#L1150).

I have a fix in my version of pdtable which does not fail on unknown method values: there is no need to support everything, only those where we can meaningfully propagate metadata. The warnings are mostly intended for development to identify situations where metadata might silently be dropped. Remember that "metadata" here includes column units, etc.

## Proposed solution 
If you revert the edits you made to `__finalize__` et al.  here, I will submit a PR with the fixes from my version once I get all the renames to line up.

File: pdtable/table_metadata.py (hunk 1 above)
Comment (jfcorbett) [reply pdtable/table_metadata.py:53]: I brought this up with @thpetsen yesterday. The answer is, there is no real use for `col_specs` here. It's `ColumnMetadata` from the old `startables` package. Its contents should be plucked out and redistributed where they belong in this new `pdtable` framework.

**Comment by jfcorbett** [comment]
@JanusWesenberg Please go right ahead and submit a PR to master with your new `__finalize__()`. We can then merge that at leisure into the branch that @thpetsen is working on.

**Pull request closed without merging**